### PR TITLE
Update response codes - Closes #504

### LINKED ISSUE #504: Update response codes
### Actual behavior

- Some param inputs throw error and results in HTTP `500` that should return a regular user input error or `404`.

- Consider the following warning

```
2021-05-06T12:21:45.717 WARN [queue] indexAccountsQueue Job failed,Cannot read property 'delegate' of undefined
2021-05-06T12:21:45.717 WARN [queue] indexAccountsQueue Job failed,TypeError: Cannot read property 'delegate' of undefined
    at /Users/michal/work/service-oss/services/core/shared/core/compat/sdk_v5/accounts.js:61:35
    at Array.forEach (<anonymous>)
    at Queue.indexAccounts (/Users/michal/work/service-oss/services/core/shared/core/compat/sdk_v5/accounts.js:60:11)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

- Account response returns delegates when `isDelegate=false`

### Expected behavior

Make sure the HTTP error codes are returned correctly.

### Steps to reproduce

`/api/v2/transactions?senderUsername=<username>&nonce=<nonce>` => `500`, should be `404`
`/api/v2/transactions?senderPublicKey=<publicKey>&nonce=<nonce>` => `500`, should be `404`
`/api/v2/accounts?isDelegate=false&status=active` => returns list of delegates, should be `404`

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #504 

### How was it solved?

- [x] Add 'exceptions.js` for consistent handling.
  - [x] /api/v2/transactions?senderUsername=<username>&nonce=<nonce> => 400
  - [x] /api/v2/transactions?senderPublicKey=<publicKey>&nonce=<nonce> => 200, 404
  - [x] /api/v2/accounts?isDelegate=false&status=active => 404
- [x] Update controller to handle error exceptions
- [x] Refactor transaction logic for:
  - [x] Code cleanup
  - [x] logical response when query with `transactionId` with offset
- [x] Refactor account logic to:
  - [x] logical response when query with `address/publicKey/Username` with offset
- [x] Refactor block logic to:
  - [x] logical response when query with `blockId` with offset
- [x] Fix `multisignature` members
- [x] `get.transactions` RPC invocation with invalid request param combinations returns INVALID_PARAMS 

### How was it tested?
- [x] Local
- [x] Jenkins

### CODE REVIEW

File: services/core/methods/api_v2/controllers/blocks.js [hunk 1]
```diff
@@ -15,52 +15,61 @@
...
-		params = remParams;
+		if (params.address) {
+			const { address, ...remParams } = params;
+			params = remParams;
 
-		params.generatorPublicKey = await Core.getPublicKeyByAny(address);
-		if (!params.generatorPublicKey) return { status: NOT_FOUND, data: { error: `Account ID ${address} not found.` } };
-	}
+			params.generatorPublicKey = await Core.getPublicKeyByAny(address);
+			if (!params.generatorPublicKey) return { status: NOT_FOUND, data: { error: `Account ID ${address} not found.` } };
```
Comment (sameersubudhi) [inline services/core/methods/api_v2/controllers/blocks.js:44]: Same as above.

File: services/core/methods/api_v2/controllers/blocks.js [hunk 2]
```diff
@@ -15,52 +15,61 @@
...
 
-	const response = await Core.getBlocks(params);
+		const response = await Core.getBlocks(params);
 
-	if (typeof params.blockId === 'string') {
-		if (isEmptyArray(response.data)) {
-			return { status: NOT_FOUND, data: { error: `Block ID ${params.blockId} not found.` } };
+		if (typeof params.blockId === 'string') {
+			if (isEmptyArray(response.data)) {
+				return { status: NOT_FOUND, data: { error: `Block ID ${params.blockId} not found.` } };
```
Comment (sameersubudhi) [inline services/core/methods/api_v2/controllers/blocks.js:51]: Same as above.

File: services/core/methods/api_v2/controllers/blocks.js [hunk 3]
```diff
@@ -15,52 +15,61 @@
...
+			}
 		}
-	}
 
-	if (typeof params.height === 'string') {
-		if (isEmptyArray(response.data)) {
-			return { status: NOT_FOUND, data: { error: `Height ${params.height} not found.` } };
+		if (typeof params.height === 'string') {
+			if (isEmptyArray(response.data)) {
+				return { status: NOT_FOUND, data: { error: `Height ${params.height} not found.` } };
```
Comment (sameersubudhi) [inline services/core/methods/api_v2/controllers/blocks.js:57]: Same as above.

File: services/core/shared/core/accounts.js [hunk 4]
```diff
@@ -28,7 +28,7 @@ const getAccounts = async params => {
 	const { status, ...remainingParams } = params;
 	let response = await coreApi.getAccounts(remainingParams);
 	if (status && ['active', 'banned', 'punished', 'standby'].some(item => status.includes(item))) {
-		response = await getDelegates(params);
+		response = params.isDelegate !== false ? await getDelegates(params) : { data: [] };
```
Comment (sameersubudhi) [inline services/core/shared/core/accounts.js:31]: Please consider adding the following comment.
```suggestion
		// Unless the request explicitly ignores delegates, consider all accounts call to include delegate info
		response = params.isDelegate !== false ? await getDelegates(params) : { data: [] };
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Please find a few minor observations below.

**Review by MichalTuleja — APPROVED** [review]

**Pull request merged**